tests/base.py

```diff
@@ -1,66 +1,69 @@
 import calendar
-import os
 import uuid
-from contextlib import contextmanager
 from copy import deepcopy
 from datetime import datetime, timedelta
 from hashlib import md5
-from typing import Iterator, MutableSequence, Optional, Sequence
+from typing import MutableSequence, Sequence
 
 from snuba import settings
 from snuba.clickhouse.http import JSONRowEncoder
 from snuba.consumer import KafkaMessageMetadata
-from snuba.datasets.dataset import Dataset
 from snuba.datasets.events_processor_base import InsertEvent
 from snuba.datasets.factory import enforce_table_writer, get_dataset
+from snuba.datasets.storages.factory import STORAGES
 from snuba.processor import InsertBatch, ProcessedMessage
 from snuba.redis import redis_client
 from snuba.utils.metrics.backends.dummy import DummyMetricsBackend
 from snuba.writer import BatchWriterEncoderWrapper, WriterTableRow
 from tests.fixtures import raw_event
 
 
-@contextmanager
-def dataset_manager(name: str) -> Iterator[Dataset]:
+def _setup() -> None:
+    assert (
+        settings.TESTING
+    ), "settings.TESTING is False, try `SNUBA_SETTINGS=test` or `make test`"
+
     from snuba.migrations.runner import Runner
-    from snuba.web.views import truncate_dataset
 
     Runner().run_all(force=True)
-    dataset = get_dataset(name)
-    truncate_dataset(dataset)
 
-    try:
-        yield dataset
-    finally:
-        truncate_dataset(dataset)
+    redis_client.flushdb()
+
+
+def _teardown() -> None:
+    from snuba.web.views import truncate_storage
+
+    for storage_key in STORAGES:
+        truncate_storage(STORAGES[storage_key])
+    redis_client.flushdb()
 
 
 class BaseTest(object):
-    def setup_method(self, test_method, dataset_name: Optional[str] = None):
-        assert (
-            settings.TESTING
-        ), "settings.TESTING is False, try `SNUBA_SETTINGS=test` or `make test`"
+    def setup_method(self) -> None:
+        _setup()
+
+    def teardown_method(self) -> None:
+        _teardown()
 
-        self.database = os.environ.get("CLICKHOUSE_DATABASE", "default")
-        self.dataset_name = dataset_name
 
-        if dataset_name is not None:
-            self.__dataset_manager = dataset_manager(dataset_name)
-            self.dataset = self.__dataset_manager.__enter__()
-        else:
-            self.__dataset_manager = None
-            self.dataset = None
+class BaseStorageTest(object):
```

Comment:
I am not sure I see a great value in preserving a class based organization for these tests.
It may have been useful earlier on, but it seems we could have a more effective reuse by actually moving away from classes and using fixtures instead.
Reasons:
- we should be able to set up and truncate a dataset (or a storage) in the same way for any test that uses them while now we have a mix of initialization done in stup/teardown methods in the classes and ad-hoc or fixture based initialization done in the module based tests. If we converge to one approach only, then I would argue we would go for fixtures and classes here would not be giving anything other than grouping tests together (which modules can do as well).
- By not relying on class inheritance to inherit the dataset initialization logic we would encourage people to write reusable fixtures that could be used in tests that do not extend from the same class hierarchy.

Reply:
Strongly agree with @fpacifici's comment. One of the biggest issues I have with class-based tests is that it's very easy to accidentally bring in a lot of unnecessary preparation of fixture state from a superclass (how many things that subclass `BaseTest` actually need all this stuff?), having to explicitly reference fixtures in the test definition makes things a lot more efficient (and easier to understand in my opinion.)